src/lib/transaction/transaction.ml

```diff
@@ -150,3 +150,36 @@ let check_well_formedness ~genesis_constants (t : t) =
       User_command.check_well_formedness ~genesis_constants cmd
   | Fee_transfer _ | Coinbase _ ->
       Ok ()
+
+let yojson_summary_of_command =
+  let mk_record type_ memo hash =
+    `List
+      [ `String type_
+      ; `String (Transaction_hash.to_base58_check hash)
+      ; `String (Signed_command_memo.to_string_hum memo)
+      ]
+  in
+  function
+  | User_command.Zkapp_command cmd as ucmd ->
+      mk_record "zkapp" (Zkapp_command.memo cmd)
+        (Transaction_hash.hash_command ucmd)
```

Comment:
This is a hash of the command's bin prot serialization. would this become expensive during high traffic? Can we maybe just include the fee payer for both zkapp and signed commands?

Reply:
A good question... I wanted to use unique identifier to eliminate possibility of edge cases (like same fee payer sending a few txs with the same nonce or something).

AFAIU hashing a transaction for internal logs would happen exactly two times: when it's just received and when it's accepted (or rejected) to the pool. My understanding was that although this hash might be non-free, it'd be dwarfed by other routines related to processing the transaction, and I made sure it happens constant number of times per transaction.

Reply:
> I wanted to use unique identifier to eliminate possibility of edge cases (like same fee payer sending a few txs with the same nonce or something)
You could include fee payer signature which I think would cover the case where same fee payer with same nonce but different account updates?

Reply:
>I made sure it happens constant number of times per transaction.

My concern is during high traffic, especially the impact on block production. I'm assuming the same function is called for transactions in a block?